apps/aecore/test/aec_eunit_SUITE.erl

```diff
@@ -63,3 +64,28 @@ end_per_testcase(_TC, Config) ->
 -spec iolist_to_s(iolist()) -> string().
 iolist_to_s(L) ->
     lists:flatten(io_lib:format("~s~n", [L])).
+
+%% We must be able to start an application when all applications it
+%% depends upon are started.
+application_test(Config) ->
+    App = aecore,
+    application:load(App),
+    TempDir = aec_test_utils:create_temp_key_dir(),
+    application:set_env(aecore, keys_dir, TempDir),
+    application:set_env(aecore, password, <<"secret">>),
+
+    {ok, Deps} = application:get_key(App, applications), 
+    AlreadyRunning = [ Name || {Name, _,_} <- proplists:get_value(running_apps, Config) ],
+    [ ok = application:ensure_started(Dep) || Dep <- Deps ],
+    
+    meck:new(aec_genesis_block_settings, []),
+    meck:expect(aec_genesis_block_settings, preset_accounts, 0, []),
+    ok = application:start(App),
+    timer:sleep(100),
+    ok = application:stop(App),
```

Comment:
application:stop leaves application loaded. I am considering whether application:unload(App) would fit here.

Reply:
No, not really, since we have tested that we can load it and we don't have to test that Erlang works. Unloading would just cause the next test using it to load it again and that's guaranteed to work.

Only small extra is that we could have saved application environment data that we don't pass on to next test. But I am not sure it survives stopped apps.